Disarm / Steal / Tazer Cleanup

tswood opened the pull request:
Should fix #19535.

Full list of adjustments:
* Made it so stealing is only available for non-hostile targets. Attempting to steal something while getting hit didn't make a lot of sense. Could see potentially re-adding for slight of hand during combat if reasonable scenario presented.
* Cleanup prompting around attack / disarm / steal. First two now will only y/n prompt if the target is not hostile, and the steal will properly warn the player that they may be attacked.
* Cleaned up and slight re-ordering in steal / disarm to ensure NPC's are angered and marked as attacked by the player.
* Fix typo in stealing message.
* Wrap angry message in translation wrapper and only display if seen.
* Update tazer code to make use of the new agress function

Coolthulhu commented on src/npc.cpp:
```diff
@@ -1172,17 +1172,19 @@ void npc::make_angry()
 
-    add_msg( "%s gets angry", name.c_str() );
+    if( g->u.sees( *this ) ) {
+      add_msg( _( "%s gets angry!" ), name.c_str() );
```
Inconsistent indent

Coolthulhu commented on src/melee.cpp:
```diff
@@ -1965,14 +1973,12 @@ void player::disarm( npc &target )
     if( hitspread < 0 ) {
         // this will not do damage, but trigger all miss effects and on_dodge on target
         melee_attack( target, false, no_technique_id, hitspread );
+        aggress_npc( target );
```
Why here and not above?

tswood replied:
For ordering of messages in the log. If aggressing first, we see the target get angry followed by the miss. This allows for what I hoped was a more reasonable flow where one would see the attempt made, followed by noting the NPC getting angry.

Coolthulhu commented on src/player.h:
```diff
@@ -1461,6 +1461,12 @@ class player : public Character, public JsonSerializer, public JsonDeserializer
+         * Makes the NPC considered to be attacked by the player first if not already hostile.
+         * Angers the NPC to allow for fight or flight.
+         */
+        void aggress_npc( npc &target );
```
Would be cleaner if it was a NPC function, since it doesn't care about player's stats.

tswood replied:
I had gone back and forth on this and I do agree. Based on functionality, it only examines NPC state. However, the action is triggered from the player point of view. I'll move this any try to rename accordingly.

Coolthulhu replied:
With functions like `on_dodge` and the like, the attacker is an argument.

tswood replied:
I can do something like `npc::make_assaulted_from( const player & u )` but at this time the player ref is not needed in any of the actual functionality. If the NPCs tracked opinion by player / creature instances it would be useful, but right now that isn't the case. I'm okay with leaving the signature like that if there is a way we can get around the build issue of an unused parameter.

Coolthulhu replied:
`npc::on_attacked( const Creature &c )` would be the most standard. Then inside, the argument would be used like `if( c.is_player() )`.

tswood replied:
Awesome, thanks. Will do.

BrettDong commented on src/iuse.cpp:
```diff
@@ -3892,46 +3892,43 @@ int iuse::tazer(player *p, item *it, bool, const tripoint &pos )
-        foe->hit_by_player = true;
+    if( foe != nullptr &&
+        !foe->is_enemy() &&
+        !p->query_yn( _("Really shock %s"), target->disp_name().c_str() ) ) {
```
Would be better if add a question mark at the end.

tswood commented:
Cleaned up formatting issues reported and moved the functionality to the NPC instead of the player. Wondering if `on_attacked( const Creature &attacker )` should be a virtual on Creature and then overridden specifically for NPCs like some of the other `on_*` functionality.

Coolthulhu commented:
Tiny merge conflict

tswood commented:
Closing to fix branch for PR.
